provider/mux_server_test.go

```diff
@@ -0,0 +1,80 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"github.com/RedisLabs/rediscloud-go-api/redis"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov5"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-testing/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"os"
+	"strconv"
+	"testing"
+)
+
+func TestMuxServer(t *testing.T) {
+
+	name := acctest.RandomWithPrefix(testResourcePrefix)
+	resourceName := "rediscloud_subscription.example"
+	testCloudAccountName := os.Getenv("AWS_TEST_CLOUD_ACCOUNT_NAME")
+
+	sdkProvider := New("dev")()
+	// frameworkProvider := NewFWProvider()
+
+	resource.Test(t, resource.TestCase{
+		ProtoV5ProviderFactories: map[string]func() (tfprotov5.ProviderServer, error){
+			"rediscloud": func() (tfprotov5.ProviderServer, error) {
+				// This method signature will include frameworkProvider
+				muxProviderServerCreator, err := MuxProviderServerCreator(sdkProvider)
+				if err != nil {
+					return nil, err
+				}
+				return muxProviderServerCreator(), nil
+			},
+		},
+		PreCheck: func() { testAccPreCheck(t); testAccAwsPreExistingCloudAccountPreCheck(t) },
+		CheckDestroy: func(s *terraform.State) error {
+			// TODO Not happy that this method has to know what types of resource were created.
```

Comment:
The solution for this might be to change all of the existing tests to use the muxed server, rather than just having a single test for the mux server?